api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1324,6 +1355,22 @@ private HttpResponse executePost(String bucketName, String objectName, Map<Strin
   }
 
 
+  private Map<String, String> normalizeHeaders(Map<String,String> headerMap) {
+    Map<String,String> normHeaderMap = new HashMap<String,String>();
+    for (Map.Entry<String, String> entry : headerMap.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      if (amzHeaders.contains(key.toLowerCase(Locale.US))) {
+        key = "x-amz-" + key;
+      } else if (!standardHeaders.contains(key.toLowerCase(Locale.US))
+                 && !key.toLowerCase(Locale.US).startsWith("x-amz-")) {
```

Comment:
Have `String headerKey = key.toLowerCase(Locale.US)` only once to avoid repeated case conversion.